examples/tutorial_client_events.c

```diff
@@ -0,0 +1,116 @@
+
+#include <stdio.h>
+#include <signal.h>
+#include "open62541.h"
+
+#ifdef UA_ENABLE_SUBSCRIPTIONS
+static void
+handler_events(UA_UInt32 monId, size_t nEventFields, const UA_Variant *eventFields, const void *context) {
+    printf("got an event! %d, %zu\n", monId, nEventFields);
+
+    for(size_t i = 0; i < nEventFields; ++i) {
+        printf("%s\n", eventFields[i].type->typeName);
+        if (UA_Variant_hasScalarType(&eventFields[i], &UA_TYPES[UA_TYPES_BYTESTRING])) {
+            UA_ByteString *string = (UA_ByteString *)eventFields[i].data;
+            printf("'%.*s'\n", (int)string->length, string->data);
+            for (size_t u = 0; u < string->length; ++u)
+                printf("%x", string->data[u]);
+            printf("\n");
+        } else if (UA_Variant_hasScalarType(&eventFields[i], &UA_TYPES[UA_TYPES_LOCALIZEDTEXT])) {
+            UA_LocalizedText *lt = (UA_LocalizedText *)eventFields[i].data;
+            printf("'%.*s'\n", (int)lt->text.length, lt->text.data);
+            printf("size: %zu\n", lt->text.length);
+        }
+    }
+}
+
+static UA_Boolean running = true;
+static void stopHandler(int sig) {
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "received ctrl-c");
+    running = false;
+}
+#endif
+
+int main(int argc, char *argv[]) {
+    signal(SIGINT, stopHandler);
+    signal(SIGTERM, stopHandler);
+
+    UA_Client *client = UA_Client_new(UA_ClientConfig_default);
+
+    UA_StatusCode retval = UA_Client_connect(client, "opc.tcp://uademo.prosysopc.com:53530/OPCUA/SimulationServer");
+    if(retval != UA_STATUSCODE_GOOD) {
+        UA_Client_delete(client);
+        return (int)retval;
+    }
+
+#ifdef UA_ENABLE_SUBSCRIPTIONS
+    /* Create a subscription */
+    UA_UInt32 subId = 0;
+    retval = UA_Client_Subscriptions_new(client, UA_SubscriptionSettings_default, &subId);
+    if(!subId) {
+        UA_Client_disconnect(client);
+        UA_Client_delete(client);
+        return (int)retval;
+    }
+    printf("Create subscription succeeded, id %u\n", subId);
+
+    /* Add a MonitoredItem */
+    UA_NodeId monitorThis = UA_NODEID_NUMERIC(0, 2253); // Root->Objects->Server
+    UA_UInt32 monId = 0;
+
+
+    // creating a single element array for the select clause of length 1
+    const size_t nSelectClauses = 1;
+    UA_SimpleAttributeOperand *selectClauses = (UA_SimpleAttributeOperand *)UA_Array_new(nSelectClauses, &UA_TYPES[UA_TYPES_SIMPLEATTRIBUTEOPERAND]);
+    if(!selectClauses){
+        UA_Client_Subscriptions_remove(client, subId);
+        UA_Client_disconnect(client);
+        UA_Client_delete(client);
+        return (int)UA_STATUSCODE_BADOUTOFMEMORY;
+    }
+
+    for(size_t i =0; i<nSelectClauses; ++i) {
+        UA_SimpleAttributeOperand_init(&selectClauses[i]);
+    }
+
+    selectClauses[0].typeDefinitionId = UA_NODEID_NUMERIC(0, 2041) ; // BaseEventType
+    selectClauses[0].browsePathSize = 1;
+    //selectClauses[0].browsePathSize = 2; // crashes
```

Comment:
Ah sorry ;)
I think I found the problem:

If I set two select clauses, i.e. the first one with `browsePathSize=1` and `Message` and then add another select clause (NOT a browsePath) I get both events:

```
const size_t nSelectClauses = 2;
//...
    selectClauses[1].typeDefinitionId = UA_NODEID_NUMERIC(0, 2041) ; // BaseEventType
    selectClauses[1].browsePath = (UA_QualifiedName*)UA_Array_new(selectClauses[0].browsePathSize, &UA_TYPES[UA_TYPES_QUALIFIEDNAME]);
    selectClauses[1].browsePathSize = 1; // crashes
    selectClauses[1].attributeId = UA_ATTRIBUTEID_VALUE;
    selectClauses[1].browsePath[0] = UA_QUALIFIEDNAME(0, "Severity");
```

Output:
```
got an event! 1, 2
LocalizedText
'Level exceeded'
size: 14
UInt16
got an event! 1, 2
LocalizedText
'Level exceeded'
size: 14
UInt16
```

I did not yet read though that part of the specification on how it should be defined...